standard/lexical-structure.md

```diff
@@ -1507,6 +1509,40 @@ A `#line hidden` directive has no effect on the compilation unit and line number
 
 > *Note*: Although a *PP_Compilation_Unit_Name* might contain text that looks like an escape sequence, such text is not an escape sequence; in this context a ‘`\`’ character simply designates an ordinary backslash character. *end note*
 
+### §Nullable-Directives Nullable directive
+
+The nullable directive controls the nullable context, as described below.
+
+```ANTLR
+fragment PP_Nullable
+    : PP_Whitespace? '#' PP_Whitespace? 'nullable' PP_Whitespace PP_Nullable_Action
+        (PP_Whitespace PP_Nullable_Target)? PP_New_Line
+    ;
+fragment PP_Nullable_Action
+    : 'disable'
+    | 'enable'
+    | 'restore'
+    ;
+fragment PP_Nullable_Target
+    : 'warnings'
+    | 'annotations'
+    ;
+```
+
+A nullable directive sets the denoted nullable context for subsequent lines of code, until another nullable directive overrides it, or until the end of the source code is reached. The effect of each form of nullable directive is, as follows:
+
+- `#nullable disable`: Sets the nullable context to “disabled”.
+- `#nullable enable`: Sets the nullable context to “enabled”.
+- `#nullable restore`: Restores the nullable context to the state specified by the external mechanism, if any.
+- `#nullable disable annotations`: Disables nullable annotations. Nullable warnings are unaffected.
+- `#nullable enable annotations`: Enables nullable annotations. Nullable warnings are unaffected.
+- `#nullable restore annotations`: Restores nullable annotations to the state specified by the external mechanism, if any. Warnings are unaffected.
+- `#nullable disable warnings`: Disables nullable warnings. Nullable annotations are unaffected.
+- `#nullable enable warnings`: Enables nullable warnings. Nullable annotations are unaffected.
+- `#nullable restore warnings`: Restores nullable warnings to the state specified by the external mechanism, if any. Nullable annotations are unaffected.
+
+Disabling the nullable context when it is already disabled has no effect. Likewise, enabling the nullable context when it is already enabled has no effect.
```

Comment:
`#nullable enable` has an effect if only one of annotations and warnings are currently enabled as it enables the other.

Reply:
In meeting: no need for this.